crates/vm/levm/src/precompiles.rs

```diff
@@ -343,12 +351,82 @@ pub fn increase_left_pad(result: &Bytes, m_size: usize) -> Result<Bytes, VMError
     }
 }
 
-fn ecadd(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+pub fn ecadd(
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata, 128)?;
+
+    increase_precompile_consumed_gas(gas_for_call, ECADD_COST.into(), consumed_gas)?;
+
+    let first_point_x = calldata
+        .get(0..32)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let first_point_x = BN254FieldElement::from_bytes_be(first_point_x)
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let first_point_y = calldata
+        .get(32..64)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let first_point_y = BN254FieldElement::from_bytes_be(first_point_y)
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let second_point_x = calldata
+        .get(64..96)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let second_point_x = BN254FieldElement::from_bytes_be(second_point_x)
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let second_point_y = calldata
+        .get(96..128)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let second_point_y = BN254FieldElement::from_bytes_be(second_point_y)
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    // If points are zero the precompile should not fail, but the conversion in
+    // BN254Curve::create_point_from_affine will, so we verify it before the conversion
+    let point_zero = BN254FieldElement::from(0);
+    let first_point_is_zero = first_point_x.eq(&point_zero) && first_point_y.eq(&point_zero);
+    let second_point_is_zero = second_point_x.eq(&point_zero) && second_point_y.eq(&point_zero);
```

Comment:
You can check if they are zero before converting them into BN254FieldElement, it will look better. After that, do the conversion